bundle/markov/sensors/sensors_rollout.py

```diff
@@ -310,6 +313,72 @@ def _right_camera_cb_(self, data):
         except Exception as ex:
             LOGGER.info("Unable to retrieve frame: %s", ex)
 
+class IMU(IMUInterface):
+    '''This is the observation class for the IMU
+    '''
+    def __init__(self, racecar_name, timeout=120.0):
+        """IMU sensor constructor
+
+        Args:
+            racecar_name (str): racecar name
+            timeout (float): time limit for buffer get method until error out.
+            The reason why using 120.0 as default value for sensor is
+            because virtual event dynamic spawning for sensor can take up to 60.0 seconds to be alive.
+        """
+        self.imu_buffer = utils.DoubleBuffer(clear_data_on_get=False)
+        rospy.Subscriber('/{}/imu'.format(racecar_name), imu_msg, self._imu_cb_)
+        self.sensor_type = Input.IMU.value
+        self.raw_data = None
+        self.timeout = timeout
+
+    def get_observation_space(self):
+        try:
+            return get_observation_space(Input.IMU.value)
+        except GenericError as ex:
+            ex.log_except_and_exit(SIMAPP_SIMULATION_WORKER_EXCEPTION)
+        except Exception as ex:
+            raise GenericRolloutException('{}'.format(ex))
+
+    def get_state(self, block=True):
+        try:
+            try:
+                # Make sure the first sensor message is the starting one
+                imu_data = self.imu_buffer.get(block=block, timeout=self.timeout)
+            
+            except utils.DoubleBuffer.Empty:
+                imu_data = self.imu_buffer.get(block=True, timeout=self.timeout)
+
+            imu_frame = []
+            imu_frame.append(imu_data.linear_acceleration.x)
+            imu_frame.append(imu_data.linear_acceleration.y)
+            imu_frame.append(imu_data.linear_acceleration.z)
+            imu_frame.append(imu_data.angular_velocity.x)
+            imu_frame.append(imu_data.angular_velocity.y)
+            imu_frame.append(imu_data.angular_velocity.z)
```

Comment:
Why are you using append instead of creating the list directly? It's less efficient.

Reply:
Good point. No real reason other than this being quick demo code.